exporters/zipkin/src/test/java/io/opentelemetry/exporter/zipkin/OtelToZipkinSpanTransformerTest.java

```diff
@@ -0,0 +1,300 @@
+                .putTag("otel.scope.version", "1.0.0")
+                .putTag("otel.library.name", "io.opentelemetry.auto")
+                .putTag("otel.library.version", "1.0.0")
+                .putTag(OtelToZipkinSpanTransformer.OTEL_STATUS_CODE, "OK")
+                .build());
+  }
+
+  @Test
+  void generateSpan_AlreadyHasHttpStatusInfo() {
+    Attributes attributeMap =
+        Attributes.of(
+            SemanticAttributes.HTTP_STATUS_CODE, 404L, stringKey("error"), "A user provided error");
+    SpanData data =
+        buildStandardSpan()
+            .setAttributes(attributeMap)
+            .setKind(SpanKind.CLIENT)
+            .setStatus(StatusData.error())
+            .setTotalAttributeCount(2)
+            .build();
+
+    assertThat(transformer.generateSpan(data))
+        .isEqualTo(
+            buildZipkinSpan(Span.Kind.CLIENT).toBuilder()
+                .clearTags()
+                .putTag(SemanticAttributes.HTTP_STATUS_CODE.getKey(), "404")
+                .putTag(OtelToZipkinSpanTransformer.OTEL_STATUS_CODE, "ERROR")
+                .putTag("error", "A user provided error")
+                .build());
+  }
+
+  @Test
+  void generateSpan_WithRpcTimeoutErrorStatus_WithTimeoutErrorDescription() {
+    Attributes attributeMap = Attributes.of(SemanticAttributes.RPC_SERVICE, "my service name");
+
+    String errorMessage = "timeout";
+
+    SpanData data =
+        buildStandardSpan()
+            .setStatus(StatusData.create(StatusCode.ERROR, errorMessage))
+            .setAttributes(attributeMap)
+            .setTotalAttributeCount(1)
+            .build();
+
+    assertThat(transformer.generateSpan(data))
+        .isEqualTo(
+            buildZipkinSpan(Span.Kind.SERVER).toBuilder()
+                .putTag(SemanticAttributes.RPC_SERVICE.getKey(), "my service name")
+                .putTag(OtelToZipkinSpanTransformer.OTEL_STATUS_CODE, "ERROR")
+                .putTag(OtelToZipkinSpanTransformer.STATUS_ERROR.getKey(), errorMessage)
+                .build());
+  }
+
+  @Test
+  void generateSpan_WithRpcErrorStatus_WithEmptyErrorDescription() {
+    Attributes attributeMap = Attributes.of(SemanticAttributes.RPC_SERVICE, "my service name");
+
+    SpanData data =
+        buildStandardSpan()
+            .setStatus(StatusData.create(StatusCode.ERROR, ""))
+            .setAttributes(attributeMap)
+            .setTotalAttributeCount(1)
+            .build();
+
+    assertThat(transformer.generateSpan(data))
+        .isEqualTo(
+            buildZipkinSpan(Span.Kind.SERVER).toBuilder()
+                .putTag(SemanticAttributes.RPC_SERVICE.getKey(), "my service name")
+                .putTag(OtelToZipkinSpanTransformer.OTEL_STATUS_CODE, "ERROR")
+                .putTag(OtelToZipkinSpanTransformer.STATUS_ERROR.getKey(), "")
+                .build());
+  }
+
+  @Test
+  void generateSpan_WithRpcUnsetStatus() {
+    Attributes attributeMap = Attributes.of(SemanticAttributes.RPC_SERVICE, "my service name");
```

Comment:
#nit: `attributesMap` => `attributes`. Note this has several occurrences in this class.

Reply:
just carried over from the original impl, but will rename.